tests/Database/DatabaseQueryBuilderTest.php

```diff
@@ -772,13 +775,13 @@ public function testAggregateResetFollowedByGet()
 	public function testAggregateResetFollowedBySelectGet()
 	{
 		$builder = $this->getBuilder();
-		$builder->getConnection()->shouldReceive('select')->once()->with('select count("column1") as aggregate from "users"', array())->andReturn(array(array('aggregate' => 1)));
-		$builder->getConnection()->shouldReceive('select')->once()->with('select "column2", "column3" from "users"', array())->andReturn(array(array('column2' => 'foo', 'column3' => 'bar')));
+		$builder->getConnection()->shouldReceive('select')->once()->with('select count("column1") as aggregate from (select "column2", "column3", "column1" from "users") as "sub"', array())->andReturn(array(array('aggregate' => 1)));
+		$builder->getConnection()->shouldReceive('select')->once()->with('select "column2", "column3", "column1" from "users"', array())->andReturn(array(array('column2' => 'foo', 'column3' => 'bar')));
 		$builder->getProcessor()->shouldReceive('processSelect')->andReturnUsing(function($builder, $results) { return $results; });
-		$builder->from('users');
+		$builder->from('users')->select('column2', 'column3', 'column1');
```

Comment:
I moved the select to be before the aggregate call, to verify the behaviour when the query itself selects a specific set of columns. We now also select `column1` to verify that it is not added again.